boa/src/environments/mod.rs

```diff
@@ -0,0 +1,36 @@
+//! This module implements ECMAScript `Environment Records`.
+//!
+//! Environments contain the bindings of identifiers to their values.
+//! The implementation differs from the methods defined by the specification,
+//! but the resulting behavior should be the same.
+//!
+//! To make the runtime more performant, environment specific behavior is split
+//! between bytecode compilation and the runtime.
+//! While the association of identifiers to values seems like a natural fit for a hashmap,
+//! lookups of the values at runtime are very expensive.
+//! Environments can also have outer environments.
+//! In the worst case, there are as many hashmap lookups, as there are environments.
+//!
+//! To avoid these costs, hashmaps are not used at runtime.
+//! At runtime, environments are represented as fixed size lists of binding values.
+//! The positions of the bindings in these lists is determined at compile time.
```

Comment:
Here compile time refers to byte-code compilation time, right? As in not Rust compile time, I was initially confused by this and I think it might make sense to make the difference clear.

Reply:
Yeah, I think I wrote "bytecode copile time" in some other places. Probably also good to write that here.

Reply:
I adjusted the docs to make this clear.